eventuals/type-check.h

```diff
@@ -19,8 +21,8 @@ struct _TypeCheck {
   template <typename Arg, typename K>
   auto k(K k) && {
     static_assert(
-        std::is_same_v<T_, ValueFrom<Arg>>,
-        "Failed to type check; expecting type on left, found type on right");
+        std::is_assignable_v<T_, ValueFrom<Arg>> || std::is_convertible_v<ValueFrom<Arg>, T_>,
```

Comment:
Refactor this to be not more then 80 characters please =)

try this with `std::disjunction_v<...>`

Reply:
Great tip, thanks! Done.